### PR TITLE
Fixes v10 JSS and SXA web.config transforms, examples on installing NodeJS in Integrated Mode

### PR DESCRIPTION
**Pull request by vtml**
# Pull Request Checklist

## Image Details

- [ ] I have added NEW image tags
- [ ] I have made significant changes to existing images

## Pull Request Details

**What this PR does / why we need it**:
NodeJS is not installed to the JSS CM / CD Docker containers, which causes integrated mode to fail.
Web.config transform was not run for JSS containers, and SXA CD containers

**Which issue(s) this PR fixes**:

Fixes https://github.com/Sitecore/docker-images/issues/386
Fixes https://github.com/Sitecore/docker-images/issues/387

**Special notes for your reviewer**:
This PR is a replacement of https://github.com/Sitecore/docker-images/pull/416 with squashed commits

I have successfully imported the Sitecore JSS React Sample application into the CM SXA+JSS container. Experience Editor is also functioning as it should.

I have created a new test folder for v10.0.0 XM1 with JSS latest and that is the test compose files I have used. I originally copied this from the XP1 compose folders and have realised there are a few small issues in that compose files. I will raise an issue to get this fixed once this PR is merged.

## Pre-submit Checklist

- [x] I have read and am familiar with the contents of [CONTRIBUTING.md](https://github.com/Sitecore/docker-images/blob/master/CONTRIBUTING.md)
- [x] I have built the images locally (all relevant OS versions, including Linux if applicable)

  - Include the `.\build.ps1` command used (don't forget to exclude your credentials!)
```powershell
.\Build.ps1 -InstallSourcePath -SitecoreUsername "fdsa" -SitecorePassword "fdsa" -OSVersion @("2004", "1909", "ltsc2019") -IncludeSpe -IncludeSxa -IncludeJss -SkipExistingImage
```
  - Ensure the build was done on a clean system (`docker system prune -af`)

- [x] I have added or updated the necessary docker-compose file(s) in the `build/windows/tests` folder
- [ ] I have updated the documentation using `build\update-documentation.ps1`
- [x] I have updated the `build\CHANGELOG.md` with details about the change(s) included

### CODE REVIEW

**Comment by jeanfrancoislarente** [comment]
Please remove any changes you may have made to `sitecore-tags.md` - this is a Sitecore-only modified file.

**Comment by vtml** [comment]
> Please remove any changes you may have made to `sitecore-tags.md` - this is a Sitecore-only modified file.

Done

File: build/windows/tests/10.0.x/build/jss-integrated-mode/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+FROM $BASE_IMAGE as base
+
+ARG NODEJS_VERSION
+
+SHELL ["powershell", "-Command", "$ErrorActionPreference = 'Stop'; $ProgressPreference = 'SilentlyContinue';"]
+
+RUN New-Item -Path 'C:\\inetpub\\wwwroot\\temp\\install\\setup' -ItemType 'Directory' -Force | Out-Null; `
+    & curl.exe -sS -L -o C:\\inetpub\\wwwroot\\temp\\install\\setup\\node.msi "https://nodejs.org/download/release/v$Env:NODEJS_VERSION/node-v$Env:NODEJS_VERSION-x64.msi";
+
+RUN $env:INSTALL_TEMP = 'C:\\inetpub\\wwwroot\\temp\\install'; `
```
Comment (jeanfrancoislarente) [inline build/windows/tests/10.0.x/build/jss-integrated-mode/Dockerfile:16]: Can these two instructions be combined into one RUN statement?

Comment (vtml) [reply build/windows/tests/10.0.x/build/jss-integrated-mode/Dockerfile]: Done

**Comment by vtml** [comment]
@jeanfrancoislarente , as discussed, this is now ready to test with a different set of compose files for XM1

**Comment by vtml** [comment]
Apologies with all the commits @jeanfrancoislarente, I was in some merge conflict hell with another branch. This PR is now ready to review.

**Pull request merged**